File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> stack = new Stack();
        StringBuilder sb = new StringBuilder();
        
        for (int i = s.length()-1; i >= 0; i--) {
            if (stack.isEmpty() ||stack.peek() != s.charAt(i)) {
                stack.push(s.charAt(i));
            }
            else {
                stack.pop();
            }
        }
        
        while (!stack.isEmpty()) {
            sb.append(stack.pop());
        }
        
        return sb.toString();
    }
}

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution(object):
    def removeDuplicates(self, s):
        """
        :type s: str
        :rtype: str
        """
        ans = []
        for char in s:
            if not ans or ans[-1] != char:
                ans.append(char)
            else:
                ans.pop()
        return "".join(ans)
        